### ISSUE TITLE
Misc. bug: convert_hf_to_gguf: Phi3 LongRoPE factors and attn_factor silently dropped for checkpoints using the new rope_parameters config schema

### ISSUE THREAD

**Issue by b-bargabus** [ggml-org/llama.cpp#24820]
### Name and Version

convert_hf_to_gguf.py with conversion/phi.py upstream tag b9728 (commit fabde3b)

### Operating systems

Linux

### Command line

```shell
python convert_hf_to_gguf.py /path/to/phi4-mini-ft-merged --outfile phi4-mini-ft.f16.gguf --outtype f16

python gguf-py/gguf/scripts/gguf_dump.py --no-tensors phi4-mini-ft.f16.gguf | grep -iE 'attn_factor|rope_factors'
```

### Problem description & steps to reproduce

conversion/phi.py (the Phi3 conversion class) reads the LongRoPE scaling from the legacy rope_scaling config key and returns early if that key is absent. Checkpoints saved with the newer rope_parameters config schema (transformers 5.x) do not have a top-level rope_scaling key, so the converter silently skips writing attn_factor and both rope_factors_long and rope_factors_short. The resulting GGUF is a LongRoPE model with no scaling tensors and no attention factor, which is mis-scaled at inference. The skip is silent, with no warning or error.

The same class already reads rope_theta from the normalized self.rope_parameters dict, so the bug is an inconsistency: theta comes from the normalized location, but the factors come from the raw legacy key.

Steps to reproduce:

Take a Phi-3 / Phi-4 checkpoint whose config.json uses the rope_parameters schema and has no top-level rope_scaling key (any checkpoint saved or re-saved by transformers 5.x, for example a merged LoRA fine-tune).
Convert it with convert_hf_to_gguf.py.
Inspect the GGUF metadata.

Result: phi3.rope.scaling.attn_factor, rope_factors_long.weight, and rope_factors_short.weight are all absent. Converting the original Microsoft checkpoint (older rope_scaling schema) produces them correctly, which masks the bug.

Root cause

conversion/base.py normalizes both schemas into a single dict:

```python
self.rope_parameters = self.hparams.get("rope_parameters", self.hparams.get("rope_scaling")) or {}
```

conversion/phi.py reads rope_theta from that normalized dict (correct):

```python
self.gguf_writer.add_rope_freq_base(self.rope_parameters.get("full_attention", self.rope_parameters)["rope_theta"])
```

but generate_extra_tensors reads the factors from the raw legacy key and returns early when it is missing:

```python
def generate_extra_tensors(self) -> Iterable[tuple[str, Tensor]]:
    ...
    # write rope scaling for long context (128k) model
    rope_scaling = self.find_hparam(['rope_scaling'], True)
    if rope_scaling is None:
        return
    ...
    self.gguf_writer.add_rope_scaling_attn_factors(attn_factor)
    ...
    yield (self.format_tensor_name(gguf.MODEL_TENSOR.ROPE_FACTORS_LONG),  torch.tensor(long_factors,  dtype=torch.float32))
    yield (self.format_tensor_name(gguf.MODEL_TENSOR.ROPE_FACTORS_SHORT), torch.tensor(short_factors, dtype=torch.float32))
```

For a rope_parameters-schema checkpoint, find_hparam(['rope_scaling']) returns None, the function returns at the early return, and the attn_factor plus both factor tensors are never written.

Proposed fix

Source the scaling in generate_extra_tensors from the already-normalized self.rope_parameters (the same dict used for rope_theta on the add_rope_freq_base line), rather than from find_hparam(['rope_scaling']). The long_factor, short_factor, and rope_type (or type) keys are present in that dict for both schemas.

As a defensive improvement, when a LongRoPE-class config is detected but the factors cannot be found, raise or warn instead of returning silently, so a future schema change surfaces loudly instead of producing a quietly mis-scaled model.

Impact

The converted model lacks the LongRoPE attention factor and per-dimension factors entirely, so positions are mis-scaled at inference. This is independent of, and compounds with, the inference-side factor-selection issue (I'm reporting that bug separately). Official Microsoft Phi checkpoints use the older rope_scaling schema and are unaffected, which is why this only appears with checkpoints produced by newer transformers versions.

### First Bad Commit

long standing, not bisected

### Relevant log output

<details>
<summary>Logs</summary>

```console

```
</details>

**Comment by b-bargabus** [comment]
Thank you!
